diff --git a/brush-parser/src/parser/peg.rs b/brush-parser/src/parser/peg.rs
index 859b051..ca153f9 100644
--- a/brush-parser/src/parser/peg.rs
+++ b/brush-parser/src/parser/peg.rs
@@ -138,7 +138,7 @@ peg::parser! {
             }
 
         rule collect_command() -> ast::CollectCommand =
-            start:specific_word("collect") arguments:(!io_redirect() w:word() { w })* redirects:redirect_list()? {
+            start:specific_word("collect") arguments:(!io_redirect() w:word() { w })* redirects:redirect_list()? !word() !io_redirect() {
                 let end = redirects
                     .as_ref()
                     .and_then(SourceLocation::location)
